### PR TITLE
[feat] add eth sdk and example

### PR DESCRIPTION
**Pull request by beepsaidblue**

### CODE REVIEW

File: core/sdk/src/factory/conveyor/eth/blockchain/index.ts [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+    const provider = new ethers.providers.Web3Provider(metamaskProvider, 'any');
+    await provider.send('eth_requestAccounts', []);
+    const providerAddress = await provider._getAddress(address);
+    return {
+      provider,
+      providerAddress
+    };
+  };
+
+  getNetworkInfo = async (networkId: number) => {
```
Comment (RustySol) [inline core/sdk/src/factory/conveyor/eth/blockchain/index.ts:66]: Network info related data should be constants in SDK, no need to call backend for data

File: core/sdk/src/factory/conveyor/eth/blockchain/index.ts [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+    return await this.networkService.getById(networkId);
+  };
+
+  getSigner = async () => {
+    const provider = new ethers.providers.Web3Provider(window.ethereum, 'any');
+    await provider.send('eth_requestAccounts', []);
+    return provider.getSigner();
+  };
+
+  private getProvider = async (basePublicParams: BlockchainPublicParams): Promise<{ provider: BlockchainProvider }> => {
```
Comment (RustySol) [inline core/sdk/src/factory/conveyor/eth/blockchain/index.ts:76]: ditto

File: core/sdk/src/factory/conveyor/eth/blockchain/index.ts [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+    return {
+      provider: new ethers.providers.JsonRpcProvider({
+        url: fullNetworkInfo.rpcUrl
+      })
+    };
+  };
+
+  private getProviderAndSigner = async (basePrivateParams: BlockchainPrivateParams) => {
+    switch (basePrivateParams.access.type) {
+      case BlockchainAccountAccessType.PrivateKey: {
```
Comment (RustySol) [inline core/sdk/src/factory/conveyor/eth/blockchain/index.ts:87]: Should remove PrivateKey

File: core/sdk/src/factory/conveyor/eth/blockchain/index.ts [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+  };
+
+  private getCreateShopSignData = (shop: CreateShopInterface) => {
+    return {
+      domain: {},
+      types: CreateShopTypes,
+      value: {
+        name: shop.name,
+        ownerAddress: shop.ownerAddress,
+        logoUrl: shop.logoUrl,
```
Comment (RustySol) [inline core/sdk/src/factory/conveyor/eth/blockchain/index.ts:120]: Remove logoUrl for v1

File: core/sdk/src/factory/conveyor/eth/blockchain/index.ts [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+    return await signer._signTypedData(domain, types, value);
+  };
+
+  signDeleteShopAction = async (baseParams: BlockchainPrivateParams, shop: DeleteShopInterface): Promise<string> => {
+    const { domain, types, value } = this.getDeleteShopSignData(shop);
+    const { signer } = await this.getProviderAndSigner(baseParams);
+    return await signer._signTypedData(domain, types, value);
+  };
+
+  signOrderConsumption = async (
```
Comment (RustySol) [inline core/sdk/src/factory/conveyor/eth/blockchain/index.ts:166]: should query by orderUuid instead, need to add backend changes for this one as well

File: core/sdk/src/factory/conveyor/eth/blockchain/index.ts [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+  private getFinalCancelOrderTransactionData = async (orders: BlockchainOrderInterface[], seaport: Seaport) => {
+    const orderComponents = orders.map(this.seaportHelper.convertOrderToOrderComponents);
+    return {
+      contractAddress: seaport.contract.address,
+      domain: await this.seaportHelper.getDomainData(seaport),
+      functionName: 'cancel',
+      params: orderComponents
+    };
+  };
+  getBlockchainPrivateKeyAccess = (privateKey: string): BlockchainPrivateKeyAccess => {
```
Comment (RustySol) [inline core/sdk/src/factory/conveyor/eth/blockchain/index.ts:213]: remove

File: core/sdk/src/factory/conveyor/eth/blockchain/index.ts [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+    const originalValue = assetInstance.value;
+    const value = convertValue ? ethers.utils.parseEther(originalValue) : originalValue;
+    return {
+      contractAddress: assetInstance.asset.address,
+      functionName: 'approve',
+      params: [spenderAddress, value]
+    };
+  };
+
+  getBlockchainOrderByConsumptionUuid = async (uuid: string) => {
```
Comment (RustySol) [inline core/sdk/src/factory/conveyor/eth/blockchain/index.ts:267]: Change this to by `OrderUuid`, need to add changes on backend for this as well

File: core/sdk/src/factory/conveyor/eth/blockchain/index.ts [hunk 8]
```diff
@@ -0,0 +1,402 @@
...
+
+  makeOrderOfferAllowance = async (
+    baseParams: BlockchainPrivateParams,
+    order: BlockchainOrderInterface,
+    consumptionUuid: string
+  ): Promise<BaseTransactionResult> => {
+    const { provider, signer } = await this.getProviderAndSigner(baseParams);
+    const contract = this.getNFTContract(order.offer, provider).connect(signer);
+    const response = await ApiCaller.request(
+      `/order/consumption/${consumptionUuid}/offer/allowance`,
```
Comment (RustySol) [inline core/sdk/src/factory/conveyor/eth/blockchain/index.ts:313]: change to `orderUuid`, need to add changes on backend for this as well

File: core/sdk/src/factory/conveyor/eth/index.ts [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+
+export class EthereumSDK implements EthereumSDKInterface {
+  private blockchainService = new BlockchainService();
+
+  makeConsumptionAllowance = async (
+    metamaskProvider: string,
+    consumptionUuid: string,
+    address: string
+  ): Promise<BaseTransactionResult> => {
+    const result = await ApiCaller.request('/order/consumption/${consumptionUuid}/allowance', RequestMethod.Get, {});
```
Comment (RustySol) [inline core/sdk/src/factory/conveyor/eth/index.ts:17]: change to orderUuid

File: core/sdk/src/factory/conveyor/eth/network/index.ts [hunk 10]
```diff
@@ -0,0 +1,30 @@
+import { ApiCaller } from '../api';
```
Comment (RustySol) [inline core/sdk/src/factory/conveyor/eth/network/index.ts:1]: all the constants should be laid out in this file then we can remove the api calls

**Comment by haVincy** [comment]
Close as we combine all the work in Minh's PR.

**Pull request closed without merging**